wolfcrypt/src/asn.c

```diff
@@ -1482,6 +1490,10 @@ int GetASN_Items(const ASNItem* asn, ASNGetData *data, int count, int complete,
         }
         /* Move index to start of length. */
         idx++;
+
+        /* Set extra length if indefinite length encoding. */
+        extra = (input[idx] == 0x80 ? 2 : 0);
```

Comment:
Change extra to indefExtra.
Indefinite length encoding was not at all handled in original implementation.
'len' will not be correct.
Should be calculated by seeing where the EOC item starts. Best to add check when EOC item (0x00, 0x00) is seen and put length in to the indef length item.
indefExtra is still needed I far as I can see.

Reply:
Use ASN_INDEF_LENGTH instead of 0x80.

Reply:
Hi @SparkiDev!  I believe I let the scope of the fix grow beyond just avoiding the out-of-bound access mentioned in the ZenDesk ticket, to start touching on adding support for indefinite-length BER encoding.  I will pull the scope of the changes here back to just avoiding the OoB access (i.e. erroring out instead), and leave the addition of indefinite-length encoding for a later effort (by those who are more in-tune with the ASN.1 logic!).   

I'll leave in as many of your suggested changes as I can without going down the indef-len path.  Thank-you for your time and guidance.  I plan to have this (Draft) PR updated shortly with minimal, OoB-focused changeset.